### ISSUE TITLE
feat: improve the Creators list loading

### ISSUE THREAD

**Issue by yurijmikhalevich** [move-fast-and-break-things/movefastandbreakthings.club#35]
We need to:
- turn the list into a carousel
- hardcode the list of 9 creators in HTML
- when the page is loaded, fetch from GH API the list of all people who have merged at least one PR into any of the repositories in the GH organization
- populate the hardcoded list of 9 creators with the additional creators fetched from GitHub

This way, fetching the list of "all creators" will be smooth (without the jumpiness that we have now).

**Comment by lord-cyclone100** [comment]
@yurijmikhalevich 
If I only turn the list into a carousel, would you assign this to me?
I am not much confident with API's since I am a beginner

**Comment by Absterrg0** [comment]
Hey @yurijmikhalevich  i could work on this issue, also what info do you need with the github accounts being called by API, as the website currently has the name,profile pic and a small description so i assume those will be the details you would want?

**Comment by Dishika18** [comment]
@yurijmikhalevich 
If no one is working on this issue, then please assign this to me under Hackoberfest 2024.

**Comment by naaa760** [comment]
Hello ! @yurijmikhalevich :)
Please assign this issue to me!
Thankyou!

**Comment by yurijmikhalevich** [comment]
@lord-cyclone100, assigned the issue to you :)

@Dishika18, @naaa760, thank you for reaching out :) I have assigned the issue to @lord-cyclone100 because they reached out first. Can you please check out other issues in the list and see if you want to take anything: https://github.com/move-fast-and-break-things/mfbt.community/issues?

**Comment by lord-cyclone100** [comment]
@yurijmikhalevich 
can i use bootstarp for the carousel?

**Comment by yurijmikhalevich** [comment]
@lord-cyclone100, I would prefer if we avoided bringing in bootstrap for the carousel. We can do something simple ourselves, or bring-in a smaller carousel-only library 💭

**Comment by yurijmikhalevich** [comment]
@lord-cyclone100, I have unassigned you from the issue. Let me know if you still want to work on it.

**Comment by AbhinavOC24** [comment]
Hey,id like to work on this

**Comment by yurijmikhalevich** [comment]
@AbhinavOC24, thank you for reaching out! Assigned! Let me know if I can help!

**Comment by AbhinavOC24** [comment]
> [@AbhinavOC24](https://github.com/AbhinavOC24), thank you for reaching out! Assigned! Let me know if I can help!


I’ve finished the carousel part, but before moving forward I want to confirm: right now the creator list is populated by org members. Instead, i need to hardcode org members and  populate it with all contributors who have merged at least one PR into any of the organization’s repositories ?

**Comment by yurijmikhalevich** [comment]
@AbhinavOC24, thank you for doing this! I think fetching just members is fine for now.

I assume fetching contributors to all projects will be:
- fetching projects
- fetching contributors for each

Which will lead to use sending more requests to GitHub, which may hit their public API ratelimits quickly.

It would be great to fetch all contributors – maybe, we can have 2 separate lists; "members" and "creators" for that. But before we do it would be great to make the website prerendered using static site generator so we don't send requests to GitHub on every page open (which is a separate ticket).

**Comment by AbhinavOC24** [comment]
Hey, quick clarification — when you mentioned “hardcode the list of 9 creators in HTML”, which 9 creators are you referring to exactly?

Right now I’ve populated it with 9 placeholder entries + org members, so the carousel is working smoothly. Should I go ahead and open a PR with this?

**Comment by yurijmikhalevich** [comment]
@yurijmikhalevich, sorry about the delay with the response! Yes, let's go! 🚀  You can also pick people from the existing creators list, so the carousel has some real data on load.

**Comment by AbhinavOC24** [comment]
Hi @yurijmikhalevich , I’ve opened a PR for this: #80 

I did what I could - added the hardcoded creators list and turned it into a carousel.  
I didn’t include dynamic PR mergers yet since that would have required multiple GitHub API requests and was hitting rate limits.

**Comment by yurijmikhalevich** [comment]
Hi @AbhinavOC24! Thank you for the update! 🙌 I see that the carousel doesn't look good on mobile screens. Also, the left/right buttons do nothing when the carousel is in the leftmost or rightmost positions. Let's either hide the buttons when they do nothing or make the carousel cyclic so the buttons always work. Can you please fix that before we can merge it? Thank you!
